### PR TITLE
Add certification filter and certified badge in helm catalog

### PR DESCRIPTION
**Pull request by rohitkrai03**
**Fixes**: https://issues.redhat.com/browse/ODC-5745

**Analysis / Root cause**: Helm certification program needs a way to add a source filter that can filter helm charts based on certification level. 

**Solution Description**:
- Added a new source filter in helm catalog based on `helm-chart.openshift.io/providerType`.
- Adda new `title` property in `CatalogItem` to render `ReactNode` when needed instead of plain `name`. We still need `name` for things like filtering and searching.
- Show a certified badge along with helm chart title in catalog tile and details panel.
- Use chart name from annotation if available.

**Screen shots / Gifs for design review**: 

@openshift/team-devconsole-ux @serenamarie125

https://user-images.githubusercontent.com/6041994/114767662-8c994c00-9d85-11eb-8b57-1cba23e6162d.mov

**Test setup:**
- Add a new helm chart repo that contains `providerType` annotations in `index.yaml`.
```yaml
apiVersion: helm.openshift.io/v1beta1
kind: HelmChartRepository
metadata:
  name: redhat-certified
spec:
  connectionConfig:
    url: >-
      https://raw.githubusercontent.com/rohitkrai03/redhat-helm-charts/certification
  name: Red Hat Certification Charts
```
- Disable the default Red Hat helm chart repo.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

### CODE REVIEW

**Review by serenamarie125 — APPROVED** [review]
This looks good to me!  Although operatorhub is not yet utilizing this pattern, we can align with Marketplace and adjust later if needed.

File: frontend/packages/helm-plugin/src/catalog/utils/const.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Annotation for human readable chart name, ready to be displayed in UI
+export const CHART_NAME_ANNOTATION = 'helm-chart.openshift.io/name';
+// Annotation for chart category, e.g. redhat, partner, community
```
Comment (serenamarie125) [inline frontend/packages/helm-plugin/src/catalog/utils/const.ts:3]: Approved too soon, where are we getting "redhat" - can we make this "Red Hat" ?

**Review by serenamarie125 — CHANGES_REQUESTED** [review]

File: frontend/packages/helm-plugin/src/catalog/utils/const.ts (hunk 1 above)
Comment (christianvogt) [reply frontend/packages/helm-plugin/src/catalog/utils/const.ts:3]: I think the well known values should be translated. Any other unknown value can be used as is.
This localize `partner` and `community` and also address the `redhat` => `Red Hat`

File: frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx [hunk 2]
```diff
@@ -23,14 +25,34 @@ export const normalizeHelmCharts = (
...
         const encodedChartURL = encodeURIComponent(chartURL);
         const href = `/catalog/helm-install?chartName=${name}&chartRepoName=${chartRepoName}&chartURL=${encodedChartURL}&preselected-ns=${activeNamespace}`;
 
+        const title =
+          providerType === 'partner' ? (
+            <>
+              {displayName} <img src={certifiedIcon} alt={t('helm-plugin~Certified')} />
+            </>
+          ) : (
+            ''
```
Comment (christianvogt) [inline frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx:53]:
```suggestion
            null
```

File: frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx [hunk 3]
```diff
@@ -88,6 +111,7 @@ export const normalizeHelmCharts = (
             name,
             chartRepositoryTitle,
             version,
+            providerType: _.capitalize(providerType),
```
Comment (christianvogt) [inline frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx:114]: We should localize the well known values.

**Comment by christianvogt** [comment]
@rohitkrai03 the vertical alignment looks off:
image
image

File: frontend/packages/helm-plugin/src/catalog/utils/const.ts (hunk 1 above)
Comment (rohitkrai03) [reply frontend/packages/helm-plugin/src/catalog/utils/const.ts:3]: Have updated the PR to use localised string for well known values and capitalize the rest.

File: frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx (hunk 2 above)
Comment (rohitkrai03) [reply frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx]: Done.

File: frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx (hunk 3 above)
Comment (rohitkrai03) [reply frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx]: Done.

**Comment by christianvogt** [comment]
Tested fine.
@rohitkrai03 just need to address the alignment issue

**Comment by rohitkrai03** [comment]
@christianvogt Updated the title to use `verticalAlign: 'middle'`.

**Comment by christianvogt** [comment]
Tested changes using test helm repository supplied.
Badge shows up as expected. Non-certified helm charts are not affected. Other catalogs continue to function.
Tested alignment on FF and Edge.

image

/label qx-approved

**Comment by christianvogt** [comment]
/lgtm
/approved
/hold

@Preeticp @pmkovar 
epic already has docs-ack - please review for docs-approved label

**Comment by Preeticp** [comment]
This issue maps to the Epic for which docs have a must have doc issue, going ahead and marking this as approved.

**Pull request merged**